libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -246,119 +240,140 @@ export class DiscoveryNodeSelector implements DiscoveryNodeSelectorService {
    * @returns a healthy discovery node endpoint
    */
   private async select() {
-    this.debug('Selecting new discovery node...')
-    const decisionTree: Decision[] = []
-
-    // Get all the services we have
-    let services = [...this.services]
-    decisionTree.push({
-      stage: DECISION_TREE_STATE.GET_ALL_SERVICES,
-      val: services
-    })
-
-    // If a whitelist is provided, filter down to it
-    if (this.config.allowlist) {
-      services = services.filter((s) => this.config.allowlist?.has(s))
-      decisionTree.push({
-        stage: DECISION_TREE_STATE.FILTER_TO_WHITELIST,
-        val: services
+    if (this.reselectLock) {
+      const prevNode = (' ' + this.selectedNode).slice(1) // Force make a copy of the string
+      await new Promise<void>((resolve) => {
```

Comment:
this works but we have to increase the max listeners because we get like 20 of these going at once sometimes. for that reason, maybe changing this to a set interval is better...

Reply:
ok that sgtm!